src/core/initialize.cpp

```diff
@@ -670,6 +671,14 @@ void on_parameter_change(int field) {
     invalidate_obs();
     recalc_forces = 1;
     break;
+  case FIELD_RIGIDBONDS:
```

Comment:
I also added dynamic switching of ghosts_have_v by counting the number of thermalized bonds.
This is done similar to Rattle, where I noticed a missing call here.